### PR TITLE
feat(stream): introduce relational top_n

### PR DESCRIPTION
**Pull request by wcy-fdu**
## What's changed and what's your intention?
This PR Implements the read/write of TopN executor by state table, which removed `flush_buffer` in `top_n_state` and `top_n_bottom_n_state`.
Since top_n has implemented a merge logic of flush_buffer and storage named `scan_and_merge`, I just refactor it and rename to `scan from relational_table` (Or we can change it to something more appropriate).

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
close https://github.com/singularity-data/risingwave/issues/1206

### CODE REVIEW

File: src/storage/src/table/state_table.rs [hunk 1]
```diff
@@ -45,13 +45,12 @@ pub struct StateTable<S: StateStore> {
     order_types: Vec<OrderType>,
 
     /// buffer key/values
-    mem_table: MemTable,
+    pub mem_table: MemTable,
```
Comment (lmatz) [inline src/storage/src/table/state_table.rs:48]: same

File: src/storage/src/table/state_table.rs [hunk 2]
```diff
@@ -45,13 +45,12 @@ pub struct StateTable<S: StateStore> {
...
 
     /// Relation layer
     cell_based_table: CellBasedTable<S>,
 
-    #[expect(dead_code)]
-    pk_indices: Vec<usize>,
+    pub pk_indices: Vec<usize>,
```
Comment (lmatz) [inline src/storage/src/table/state_table.rs:53]: same

Comment (wcy-fdu) [reply src/storage/src/table/state_table.rs:53]: Thanks for your comments and I will refactor this.

**Review by lmatz — APPROVED** [review]
LGTM!

**Pull request merged**